src/views/virtualmachines/actions/components/CloneVMModal/components/CloningStatus.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { FC, useEffect } from 'react';
+
+import { modelToGroupVersionKind } from '@kubevirt-ui/kubevirt-api/console';
+import VirtualMachineCloneModel from '@kubevirt-ui/kubevirt-api/console/models/VirtualMachineCloneModel';
+import { V1alpha1VirtualMachineClone } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Alert,
+  AlertVariant,
+  ProgressStep,
+  ProgressStepper,
+  ProgressStepVariant,
+  Stack,
+} from '@patternfly/react-core';
+
+import { CLONING_STATUSES, STATUS_TO_PROGRESS_VARIANT } from '../utils/constants';
+
+type CloningStatusProps = {
+  onComplete: (status: CLONING_STATUSES) => void;
+  vmCloneRequest: V1alpha1VirtualMachineClone;
+};
+
+const CloningStatus: FC<CloningStatusProps> = ({ onComplete, vmCloneRequest }) => {
+  const { t } = useKubevirtTranslation();
+
+  const [freshVMCloneRequest] = useK8sWatchResource<V1alpha1VirtualMachineClone>(
+    vmCloneRequest && {
+      groupVersionKind: modelToGroupVersionKind(VirtualMachineCloneModel),
+      name: vmCloneRequest.metadata.name,
+      namespace: vmCloneRequest.metadata.namespace,
+    },
+  );
+
+  const cloneRequest = freshVMCloneRequest || vmCloneRequest;
```

Comment:
why? why fresh required if u are getting the promise of vmclone?

Reply:
`vmCloneRequest` is the clone request that we get from the k8sCreate. The request changes its status over time. So with `freshVMCloneRequest`  we take track of this state. If the request fails, the user will see the failure. If the request is completed, the user will see complete at some point. 
At the beginning `freshVMCloneRequest` is undefined so for this reason I made an `|| ` with the initial clone request.